### ISSUE TITLE
`StringIO.set_encoding_by_bom()`

### ISSUE THREAD

**Issue by rurounijones** [oracle/truffleruby#3632]
Some of our code makes use of this method: https://ruby-doc.org/3.2.2/exts/stringio/StringIO.html#method-i-set_encoding_by_bom

This fails in our test suite when running TruffleRuby 24.0.2

```
     NoMethodError:
       undefined method `set_encoding_by_bom' for #<StringIO:0x848638>
```

This does not appear to be implemented in Truffleruby but`set_encoding` is : https://github.com/oracle/truffleruby/blob/3cd422433deebe3fa664f8c4540811c42ca02e93/lib/truffle/stringio.rb#L203

(As a sidenote: Love what you guys are doing with `TruffleRuby`; excited for the future)

**Comment by andrykonchin** [comment]
Thank you for the report, we'll look into it.

**Comment by andrykonchin** [comment]
Fixed in 61396dc661d67ceef5e681ba519993ee0c2f8966.
